File: new/atoms/stage-index.test.mjs
import { stageIndex as __pure_stageIndex } from './stage-index.mjs';
// צילום-מקומי + עטיפת-כריכה (מנוע-הטיהור v2; בדיקה לא מייבאת אטום-שכן)
const __d_stageIndex_AYIN_STAGES = ['new', 'lead', 'eyes', 'answer', 'done'];
const stageIndex = (...a) => __pure_stageIndex(...a, ...Array(Math.max(0, 1 - a.length)).fill(undefined), __d_stageIndex_AYIN_STAGES);
let f = 0;
const eq = (a, b, msg) => { if (a !== b) { console.error(`✗ ${msg} ⇒ ${JSON.stringify(a)}`); f = 1; } };

// 1-3) חמשת השלבים בסדרם
eq(stageIndex('new'), 0, 'new שגוי');
eq(stageIndex('lead'), 1, 'lead שגוי');
eq(stageIndex('eyes'), 2, 'eyes שגוי');
eq(stageIndex('answer'), 3, 'answer שגוי');
eq(stageIndex('done'), 4, 'done שגוי');

// 4) לא-מוכר ⇒ 0 (לא ‎-1)
eq(stageIndex('foo'), 0, 'לא-מוכר לא נפל ל-0');

if (f) process.exit(1);
console.log('✓ stage-index: 4 דוגמאות-חוזה — ירוק');

File: new/dart-maor/stage-index_test.dart
// בדיקת-חוזה (רתמת-זהב) · stageIndex — מייבאת אך ורק את האטום-שלה (חוק-4).
// כל דוגמאות-החוזה (stage-index.contract.md 1–4) ובדיקת-ה-JS
// (new/atoms/stage-index.test.mjs — 6 השוואות) מתורגמות ביט-אחר-ביט:
// אותם קלטים ⇒ אותם אינדקסים. הפלט סקלרי (int) ⇒ כלל-8 (מערכים) לא-ישים.
// הרצה: dart run --enable-asserts new/dart-maor/stage-index_test.dart ⇒ OK
import 'stage-index.dart';

void main() {
  // [קלט, פלט-מוקלט] — הועתק אחד-לאחד מ-stage-index.test.mjs.
  final cases = <List<dynamic>>[
    // 1-3) חמשת השלבים בסדרם (חוזה 1–3)
    ['new', 0],
    ['lead', 1],
    ['eyes', 2],
    ['answer', 3],
    ['done', 4],
    // 4) לא-מוכר ⇒ 0 (לא ‎-1) — חוזה 4
    ['foo', 0],
  ];

  var f = 0;
  for (final row in cases) {
    final got = stageIndex(row[0]);
    if (got != row[1]) {
      print("✗ stageIndex('${row[0]}') ⇒ $got ≠ ${row[1]}");
      f = 1;
    }
  }
  if (f != 0) throw StateError('stage-index: סטייה מדוגמאות-החוזה');
  print('✓ stage-index: ${cases.length} דוגמאות-חוזה — ירוק');
  print('OK');
}
